messaging/kafka/src/main/java/io/helidon/messaging/kafka/SimpleKafkaConsumer.java

```diff
@@ -0,0 +1,322 @@
+
+    private static final Logger LOGGER = Logger.getLogger(SimpleKafkaConsumer.class.getName());
+    private final KafkaConfigProperties properties;
+
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+    private final PartitionsAssignedLatch partitionsAssignedLatch = new PartitionsAssignedLatch();
+    private final String consumerId;
+    private ExecutorService executorService;
+    private ExecutorService externalExecutorService;
+    private final List<String> topicNameList;
+    private final KafkaConsumer<K, V> consumer;
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param channelName key in configuration
+     * @param config      Helidon {@link io.helidon.config.Config config}
+     * @see KafkaConfigProperties
+     * @see io.helidon.config.Config
+     */
+    public SimpleKafkaConsumer(String channelName, Config config) {
+        this(channelName, config, null);
+    }
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param channelName     key in configuration
+     * @param config          Helidon {@link io.helidon.config.Config config}
+     * @param consumerGroupId Custom group.id, can be null, overrides group.id from configuration
+     * @see KafkaConfigProperties
+     * @see io.helidon.config.Config
+     */
+    public SimpleKafkaConsumer(String channelName, Config config, String consumerGroupId) {
+        this.properties = new KafkaConfigProperties(config.get("mp.messaging.incoming").get(channelName));
+        this.properties.setProperty(KafkaConfigProperties.GROUP_ID, getOrGenerateGroupId(consumerGroupId));
+        this.topicNameList = properties.getTopicNameList();
+        this.consumerId = channelName;
+        this.consumer = new KafkaConsumer<>(properties);
+    }
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     */
+    public SimpleKafkaConsumer(Config config) {
+        this.properties = new KafkaConfigProperties(config);
+        this.properties.setProperty(KafkaConfigProperties.GROUP_ID, getOrGenerateGroupId(null));
+        this.topicNameList = properties.getTopicNameList();
+        this.consumerId = null;
+        this.consumer = new KafkaConsumer<>(properties);
+    }
+
+    /**
+     * Execute supplied consumer for each received record.
+     *
+     * @param function to be executed for each received record
+     * @return {@link java.util.concurrent.Future}
+     */
+    public Future<?> consumeAsync(Consumer<ConsumerRecord<K, V>> function) {
+        return this.consumeAsync(Executors.newWorkStealingPool(), null, function);
```

Comment:
Creating a new executor service for each request is wrong use of executors.

Reply:
This part is out.